Change "named_guard" in native_functions to "supports_named_tensor"

zou3519 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#26352 Change "named_guard" in native_functions to "supports_named_tensor"**

"named_guard: P" is the same as "supports_named_tensor: !P".
Also changed the error message to be more understandable to users.

Test Plan:
- `TEST_NAMEDTENSOR=1 pytest test/test_namedtensor.py -v`
- [namedtensor ci]

Differential Revision: [D17426234](https://our.internmc.facebook.com/intern/diff/D17426234)

zou3519 committed:
Change "named_guard" in native_functions to "supports_named_tensor"

"named_guard: P" is the same as "supports_named_tensor: !P".
Also changed the error message to be more understandable to users.

Test Plan:
- `TEST_NAMEDTENSOR=1 pytest test/test_namedtensor.py -v`
- [namedtensor ci]

nairbv reviewed (approved):
I agree this is much clearer

zou3519 committed:
Update on "Change "named_guard" in native_functions to "supports_named_tensor""


"named_guard: P" is the same as "supports_named_tensor: !P".
Also changed the error message to be more understandable to users.

Test Plan:
- `TEST_NAMEDTENSOR=1 pytest test/test_namedtensor.py -v`
- [namedtensor ci]

Differential Revision: [D17426234](https://our.internmc.facebook.com/intern/diff/D17426234)
